cranelift/codegen/src/isa/riscv64/inst.isle

```diff
@@ -1913,6 +1914,33 @@
 (decl intcc_to_extend_op (IntCC) ExtendOp)
 (extern constructor intcc_to_extend_op intcc_to_extend_op)
 
+
+;; parameters are in_ty out_ty input is_signed  
+(decl clamp_fcvt_from_float (Type Type Reg bool) Reg)
+(rule 3 (clamp_fcvt_from_float _ $I32 r _)
+  r)
+(rule 2 (clamp_fcvt_from_float _ $I64 r _)
+  r)
+
+(rule 1 (clamp_fcvt_from_float in_ty ty r is_signed)
+   (let
+    ((min_value i32 (i8_i16_min_value ty is_signed))
+    (min_reg Reg (imm in_ty (i32_2_float_bits min_value in_ty)))
+    (tmp Reg (fmax_ in_ty min_reg r))
+    (max_value i32 (i8_i16_max_value ty is_signed))
+    (max_reg Reg (imm in_ty (i32_2_float_bits max_value in_ty))))
+    (fmin_ in_ty tmp max_reg)))
+
+(decl i8_i16_max_value (Type bool) i32)
```

Comment:
I guess I have a slight preference for having these on ISLE as well, but I'm not too sure. @jameysharp what do you think about this?

Reply:
In general I agree that I'd like to do as much in ISLE as possible.

I think there are a number of tricks we can use which might clean this up, but I'm having trouble putting them together into a complete suggestion.

The pure ISLE version of this probably would return the appropriate hex-literal float for each combination of type and is_signed. I'm not sure how easy that'll be to write, understand, or maintain.

Leaving some of the work to Rust, we can factor it differently to shorten the code. With signed conversion, we can start with `i64::MAX` or `i64::MIN`, then `x >> (64 - ty.bits())`, then use `f32_bits(x as f32)` or `f64_bits(x as f64)`. Unsigned is the same except for starting with `u64::MAX` or `u64::MIN` and using unsigned shifts. Since we apparently only need to clamp for `fits_in_16` types (why is that though?) we can replace `64` with `16` in the first two steps.

Reply:
> Since we apparently only need to clamp for fits_in_16 types (why is that though?) 

We have native instructions for this operation in both 32 and 64 bit variants, but for <16 we clamp the values to the maximum range supported by i8/i16 and then do the 32 bit conversion.